escrow/src/data_structures.sw

```diff
@@ -58,14 +58,14 @@ pub struct EscrowInfo {
 impl EscrowInfo {
     pub fn new(
         arbiter: Arbiter,
-        assets: [Asset; 2],
+        assets: Vec<Asset>,
         buyer: Identity,
         deadline: u64,
         seller: Identity,
     ) -> Self {
         Self {
             arbiter,
-            assets,
+            assets: [assets.get(0).unwrap(), assets.get(1).unwrap()],
```

Comment:
This should not be hardcoding 2 assets, it should take the entire array

Reply:
We can't use vecs in structs yet so `assets` in `EscrowInfo` needs to be a hardcoded array.  The only alternative I can think of is to move `assets` out of the struct and use it separately in storage.

Reply:
I believe we track the ID of the escrow, correct? What is the issue with moving the vector out of the struct and having an additional storage variable to index?

Reply:
Code quality I guess.  But functionality probably trumps such a nit.

Reply:
We can't use storage containers within each other yet so we can't track the storage vector as a separate storage variable inside of a storage map, so I think we are stuck with a static array inside EscrowInfo for now.

Reply:
I don't understand.

Reply:
So ideally we would like the `assets` variable inside of `EscrowInfo` to be a `StorageVec<Asset>` but [this](https://github.com/FuelLabs/sway/issues/2465) is blocking us.  A workaround is to have a separate storage variable `assets" StorageMap<u64, StorageVec<Asset>`, but [this](https://github.com/FuelLabs/sway/issues/2639) is blocking as well.

Reply:
And assets can't be a regular vec it has to be a `StorageVec`

Reply:
There is a solution but it's nasty and I'd rather not go down that path but then again I'd rather not go down this one either.

Struct keeps track of the number of assets and the index of the storagevec where the first asset is. You can then iterate from that point to get the assets.

Thoughts?

Reply:
Not a huge fan, but at this point I think it's the only option to get the contract fully functional which would be nice for the frontend and cool in general.  The Sway implementation should be pretty straightforward.  To display an Escrow's Asset information this way from the frontend perspective might be a bit of a pain although I think I could implement a temporary getter function for the contract to avoid this pain until the blockers are resolved.